pkg/controller/feature-gates.go

```diff
@@ -0,0 +1,59 @@
+package controller
+
+import (
+	"context"
+	"k8s.io/apimachinery/pkg/types"
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1beta1"
+	"kubevirt.io/containerized-data-importer/pkg/common"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sync"
+)
+
+const (
+	// SkipWaitForFirstConsumerVolumes - if enabled will not schedule worker pods on a storage with WaitForFirstConsumer binding mode
+	SkipWaitForFirstConsumerVolumes = "SkipWaitForFirstConsumerVolumes"
+)
+
+// FeatureGates is a util for determining whether an optional feature is enabled or not.
+type FeatureGates struct {
+	client               client.Client
+	lock                 *sync.Mutex
+	lasValidFeatureGates []string
```

Comment:
what is the use case for caching feature gates?  I think I would prefer an API that always checks the config object and returns an error if it cannot be obtained

Reply:
FeatureGate might be used in a loop in controllers where it is often read, but is by design rarely changed (and only from outside), so I was thinking this is ideal place to cache last valid value in case there is any error on client api/communication with k8s/ or even featureGates format etc... (and I was loosely following https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-config/config-map.go).

But I also understand that it might be good to just return error and fail this reconciliation loop, and retry again. 

Honestly I have no experience which way might behave better in big cluster.

Reply:
The CDIConfig object is already cached with CR client, so not gaining much with additional cache IMO

Reply:
Ok, so I can can try it. So now the api is (bool, error). I still cannot get used to it and to the `golang` way. I preferred the previous way, but it is not a strong preference.